types/block.go

```diff
@@ -1295,6 +1295,15 @@ type Data struct {
 	// This means that block.AppHash does not include these txs.
 	Txs Txs `json:"txs"`
 
+	// SquareSize is the size of the square after splitting all the block data
+	// into shares. The erasure data is discarded after generation, and keeping this
+	// value avoids unnecessarily regenerating all of the shares when returning
+	// proofs that some element was included in the block
+	SquareSize uint64 `json:"square_size"`
+
+	// DataRootHash is the root hash of the data square.
+	DataRootHash cmtbytes.HexBytes `json:"data_root_hash"`
```

Comment:
[question] just wanted to make sure, `block.header.Hash` is equal to this value `block.Data.DataRootHash`, correct? 

This is still the case in the v0.38 version of the code?

in v0.34 we passed the full header (https://github.com/celestiaorg/celestia-core/blob/v0.34.x-celestia/state/execution.go#L185) and used that in processProposal (https://github.com/celestiaorg/celestia-app/blob/main/app/process_proposal.go#L182).

now we pass data root hash directly - https://github.com/celestiaorg/celestia-core/blob/marko/core_changes_v3/state/execution.go#L185

Reply:
Yeah, I understand that. I would also prefer to keep similar to upstream. 
Just want to make sure that if you query for example:

Header: https://public-celestia-rpc.numia.xyz/header?height=10000
ShareProof: https://public-celestia-rpc.numia.xyz/prove_shares?height=10000&startShare=0&endShare=1

that you can still plug in the data root as: `shareProof.Validate(header.DataHash)`

Reply:
the flow should still be the same. We populate the hash in data field with a hash there fore when we call `.Hash()` it wont attempt to fill it with new info instead use the existing data.